plugin/evm/vm.go

```diff
@@ -256,6 +266,9 @@ type VM struct {
 	ethTxPushGossiper  avalancheUtils.Atomic[*gossip.PushGossiper[*GossipEthTx]]
 	ethTxPullGossiper  gossip.Gossiper
 
+	uptimeManager  uptime.PausableManager
+	validatorState validators.State
```

Comment:
could we put the additions in a separate file & ideally a separate struct? (these fields with validatorsDB & the related functions) 
This struct can get called from the VM for initialize, state change, shutdown & for connect/disconnect.
My main concerns are:
- Size of vm.go file (goes from ~1400 -> 1500) with this change,
- Having smaller structs means less tangling of initialization logic and possibly shorter UTs.

Reply:
I will try that with a separate PR (including signing changes). Yea I also think VM does too much work of dispatching the update, doing the update itself etc. So they can be moved to a manager-maintainer like struct.